File: app/src/main/java/com/example/massa/luxvilla/Actividades/MainActivity.java
package com.example.massa.luxvilla.Actividades;


import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.pm.ShortcutInfo;
import android.content.pm.ShortcutManager;
import android.graphics.drawable.Icon;
import android.net.ConnectivityManager;
import android.net.Uri;
import android.os.Build;
import android.preference.PreferenceManager;
import android.preference.SwitchPreference;
import android.provider.Settings;
import android.support.annotation.NonNull;
import android.support.design.widget.AppBarLayout;
import android.support.design.widget.NavigationView;
import android.support.design.widget.Snackbar;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.content.ContextCompat;
import android.support.v4.view.ViewPager;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.transition.ChangeBounds;
import android.transition.Transition;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.example.massa.luxvilla.R;
import com.example.massa.luxvilla.separadores.separadoraveiro;
import com.example.massa.luxvilla.separadores.separadorbraga;
import com.example.massa.luxvilla.separadores.separadorporto;
import com.example.massa.luxvilla.separadores.separadortodas;
import com.example.massa.luxvilla.sqlite.BDAdapter;
import com.example.massa.luxvilla.utils.firebaseutils;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.messaging.FirebaseMessaging;
import com.lapism.searchview.SearchAdapter;
import com.lapism.searchview.SearchHistoryTable;
import com.lapism.searchview.SearchItem;
import com.lapism.searchview.SearchView;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;

public class MainActivity extends AppCompatActivity {

    ViewPager vwpgr;
    TabLayout tbs;
    final int SEPARADOR_TODAS=0;
    final int SEPARADOR_AVEIRO=1;
    final int SEPARADOR_BRAGA=2;
    final int SEPARADOR_PORTO=3;
    BDAdapter adapter;
    SharedPreferences sharedPreferencesnotification;
    com.lapism.searchview.SearchView searchViewpr;
    List<SearchItem> sugestions;
    SearchHistoryTable msearchHistoryTable;
    AppBarLayout appBarLayout;
    NavigationView navigationView;
    DrawerLayout drawerLayout;
    TextView tvusername;
    TextView tvusermail;
    CircleImageView ivprofile;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        if( Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP ){
            Transition tr=new ChangeBounds();
            getWindow().setSharedElementExitTransition(tr);
        }
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        sharedPreferencesnotification= PreferenceManager.getDefaultSharedPreferences(MainActivity.this);
        boolean notificationstate=sharedPreferencesnotification.getBoolean(getResources().getString(R.string.notificaçãoes),true);
        if (notificationstate){
            FirebaseMessaging.getInstance().subscribeToTopic("todos");
        }else{
            FirebaseMessaging.getInstance().unsubscribeFromTopic("todos");
        }


        searchViewpr=(com.lapism.searchview.SearchView)findViewById(R.id.searchViewp);
        searchViewpr.setHint("LuxVilla: Todas");
        searchViewpr.setTextSize(18);
        searchViewpr.setVoice(true);
        searchViewpr.setTextStyle(1);
        searchViewpr.setCursorDrawable(R.drawable.cursor);
        searchViewpr.setIconColor(ContextCompat.getColor(MainActivity.this,R.color.colorPrimary));
        searchViewpr.setShouldClearOnClose(true);

        drawerLayout=(DrawerLayout)findViewById(R.id.drawerll);
        navigationView=(NavigationView)findViewById(R.id.navigationview);

        tbs=(TabLayout)findViewById(R.id.tabs);
        tbs.setSelectedTabIndicatorColor(ContextCompat.getColor(MainActivity.this,R.color.colorAccent));
        vwpgr= (ViewPager) findViewById(R.id.vpgr);
        adapter=new BDAdapter(this);


        tbs.setupWithViewPager(vwpgr);



        int numerocolunas=adapter.numerodecolunas();
        if (numerocolunas==0 && !isNetworkAvailable(this)){
            vwpgr.setVisibility(View.GONE);
            tbs.setVisibility(View.GONE);
        }else {
            adaptadorpaginas adaptador=new adaptadorpaginas(getSupportFragmentManager());
            vwpgr.setAdapter(adaptador);

            vwpgr.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
                @Override
                public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

                }

                @Override
                public void onPageSelected(int position) {

                    switch (position){
                        case 0:
                            searchViewpr.setHint("LuxVilla: Todas");
                            navigationView.getMenu().getItem(0).setChecked(true);
                            break;
                        case 1:
                            searchViewpr.setHint("LuxVilla: Aveiro");
                            navigationView.getMenu().getItem(1).setChecked(true);
                            break;
                        case 2:
                            searchViewpr.setHint("LuxVilla: Braga");
                            navigationView.getMenu().getItem(2).setChecked(true);
                            break;
                        case 3:
                            searchViewpr.setHint("LuxVilla: Porto");
                            navigationView.getMenu().getItem(3).setChecked(true);
                            break;
                    }
                    if (!isNetworkAvailable(MainActivity.this)){
                        Snackbar.make(vwpgr,"Sem ligação há internet",Snackbar.LENGTH_LONG).setAction("ligar", new View.OnClickListener() {
                            @Override
                            public void onClick(View v) {
                                Intent defenicoes = new Intent(Settings.ACTION_WIFI_SETTINGS);
                                startActivity(defenicoes);
                            }
                        }).setActionTextColor(ContextCompat.getColor(MainActivity.this,R.color.colorAccent)).show();
                    }
                }

                @Override
                public void onPageScrollStateChanged(int state) {

                }
            });
        }

        sugestions= new ArrayList<>();
        msearchHistoryTable=new SearchHistoryTable(MainActivity.this);
        msearchHistoryTable.setHistorySize(3);
        searchViewpr.setOnQueryTextListener(new com.lapism.searchview.SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextChange(String newText) {
                return false;
            }

            @Override
            public boolean onQueryTextSubmit(String query) {
                sugestions.add(new SearchItem(query));
                msearchHistoryTable.addItem(new SearchItem(query));


                Intent search=new Intent(MainActivity.this,searchableactivity.class);
                search.putExtra("query",query);
                startActivity(search);
                searchViewpr.setShouldClearOnClose(true);
                searchViewpr.close(true);
                return true;
            }
        });
        final SearchAdapter searchAdapter=new SearchAdapter(MainActivity.this,sugestions);
        searchAdapter.addOnItemClickListener(new SearchAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(View view, int position) {
                TextView textView = (TextView) view.findViewById(R.id.textView_item_text);
                String query = textView.getText().toString();

                Intent search=new Intent(MainActivity.this,searchableactivity.class);
                search.putExtra("query",query);
                startActivity(search);
                searchViewpr.close(true);
            }
        });
        searchViewpr.setAdapter(searchAdapter);
        searchViewpr.setOnMenuClickListener(new com.lapism.searchview.SearchView.OnMenuClickListener() {
            @Override
            public void onMenuClick() {
                drawerLayout.openDrawer(navigationView);
            }
        });

        appBarLayout=(AppBarLayout)findViewById(R.id.appbarll);
        searchViewpr.setOnOpenCloseListener(new SearchView.OnOpenCloseListener() {
            @Override
            public boolean onClose() {
                int id=vwpgr.getCurrentItem();
                switch (id){
                    case 0:
                        searchViewpr.setHint("LuxVilla: Todas");
                        break;
                    case 1:
                        searchViewpr.setHint("LuxVilla: Aveiro");
                        break;
                    case 2:
                        searchViewpr.setHint("LuxVilla: Braga");
                        break;
                    case 3:
                        searchViewpr.setHint("LuxVilla: Porto");
                        break;
                }
                searchViewpr.setTextStyle(1);
                return true;
            }

            @Override
            public boolean onOpen() {
                searchViewpr.setHint(getResources().getString(R.string.app_hint));
                searchViewpr.setTextStyle(0);
                return true;
            }
        });

        View headerLayout = navigationView.getHeaderView(0);
        ivprofile=(CircleImageView) headerLayout.findViewById(R.id.profileimage);
        tvusername=(TextView) headerLayout.findViewById(R.id.textviewusername);
        tvusermail=(TextView) headerLayout.findViewById(R.id.textviewusermail);


        navigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                switch (item.getItemId()){
                    case R.id.navigation_item_1:
                        vwpgr.setCurrentItem(0);
                        break;
                    case R.id.navigation_item_2:
                        vwpgr.setCurrentItem(1);
                        break;
                    case R.id.navigation_item_3:
                        vwpgr.setCurrentItem(2);
                        break;
                    case R.id.navigation_item_4:
                        vwpgr.setCurrentItem(3);
                        break;
                    case R.id.navigation_subheader_1:
                        Intent it=new Intent(MainActivity.this, settings.class);
                        startActivity(it);
                        break;
                    case R.id.navigation_subheader_profile:
                        //TODO: profile
                        startActivity(new Intent(MainActivity.this, Userprofile.class));
                        break;
                    case R.id.navigation_subheader_signout:
                        FirebaseAuth.getInstance().signOut();
                        startActivity(new Intent(MainActivity.this,Loginactivity.class));
                        finish();
                        break;
                }
                drawerLayout.closeDrawers();
                return true;
            }
        });


        ShortcutManager shortcutManager;

        ShortcutInfo shortcut;
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N_MR1) {
            shortcutManager = getSystemService(ShortcutManager.class);

            shortcut = new ShortcutInfo.Builder(this, "id1")
                    .setShortLabel(getResources().getString(R.string.busca))
                    .setLongLabel(getResources().getString(R.string.busca))
                    .setIcon(Icon.createWithResource(MainActivity.this, R.drawable.ic_shortcut_search))
                    .setIntents(
                            new Intent[]{
                                    new Intent(Intent.ACTION_MAIN, Uri.EMPTY, this, MainActivity.class).setFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK),
                                    new Intent(Intent.ACTION_MAIN, Uri.EMPTY, this, searchableactivity.class)
                            })
                    .build();
            ShortcutInfo shortcutInfoLink = new ShortcutInfo.Builder(this, "shortcut_web")
                    .setShortLabel("website")
                    .setLongLabel("LuxVilla website")
                    .setIcon(Icon.createWithResource(this, R.drawable.logo))
                    .setIntent(new Intent(Intent.ACTION_VIEW, Uri.parse("http://brunoferreira.esy.es/")))
                    .build();


            shortcutManager.setDynamicShortcuts(Arrays.asList(shortcut, shortcutInfoLink));
        }

    }


    @Override
    public void onDestroy(){
        super.onDestroy();
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);

        return true;
    }

    public boolean onOptionsItemSelected(MenuItem item) {

        return super.onOptionsItemSelected(item);
    }

    private class adaptadorpaginas extends FragmentPagerAdapter{

        adaptadorpaginas(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int position) {

            switch (position){
                case SEPARADOR_TODAS:
                    return new separadortodas();
                case SEPARADOR_AVEIRO:
                    return new separadoraveiro();
                case SEPARADOR_BRAGA:
                    return new separadorbraga();
                case SEPARADOR_PORTO:
                    return new separadorporto();
            }
            return null;
        }

        public CharSequence getPageTitle(int position){
            String[] tabsname={"TODAS","AVEIRO","BRAGA","PORTO"};

            return tabsname[position];
        }

        @Override
        public int getCount() {
            return 4;
        }
    }

    @Override
    public void onResume(){
        super.onResume();
        firebaseutils.getuserdata(MainActivity.this,tvusername,tvusermail, ivprofile);
    }

    public void onBackPressed() {

        if (drawerLayout.isDrawerOpen(navigationView)){
            drawerLayout.closeDrawer(navigationView);
        }else {
            super.onBackPressed();
        }

    }

    public boolean isNetworkAvailable(final Context context) {
        final ConnectivityManager connectivityManager = ((ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE));
        return connectivityManager.getActiveNetworkInfo() != null && connectivityManager.getActiveNetworkInfo().isConnected();
    }
}

File: app/src/main/java/com/example/massa/luxvilla/Actividades/MainActivity.kt
package com.example.massa.luxvilla.Actividades

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.content.pm.ShortcutInfo
import android.content.pm.ShortcutManager
import android.graphics.drawable.Icon
import android.net.ConnectivityManager
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.preference.PreferenceManager
import android.provider.Settings
import android.support.design.widget.Snackbar
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.content.ContextCompat
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import android.support.v7.app.AppCompatDelegate
import android.transition.ChangeBounds
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.TextView
import com.example.massa.luxvilla.R
import com.example.massa.luxvilla.separadores.separadoraveiro
import com.example.massa.luxvilla.separadores.separadorbraga
import com.example.massa.luxvilla.separadores.separadorporto
import com.example.massa.luxvilla.separadores.separadortodas
import com.example.massa.luxvilla.sqlite.BDAdapter
import com.example.massa.luxvilla.utils.firebaseutils
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.messaging.FirebaseMessaging
import com.lapism.searchview.SearchAdapter
import com.lapism.searchview.SearchHistoryTable
import com.lapism.searchview.SearchItem
import com.lapism.searchview.SearchView
import de.hdodenhof.circleimageview.CircleImageView
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.header.*
import java.util.*

/**
 * Created by massa on 26/10/2017.
 */
class MainActivity : AppCompatActivity() {

    internal val SEPARADOR_TODAS = 0
    internal val SEPARADOR_AVEIRO = 1
    internal val SEPARADOR_BRAGA = 2
    internal val SEPARADOR_PORTO = 3
    internal var adapter: BDAdapter? = null
    internal val PREFSNAME = "FAVS"
    internal var sharedPreferencesapp: SharedPreferences? = null
    internal var sharedPreferenceslikes: SharedPreferences? =null
    internal var sugestions: MutableList<SearchItem>? = null
    internal var msearchHistoryTable: SearchHistoryTable? = null
    var txtuser: TextView? = null
    var txtmail: TextView? = null
    var profileimg: CircleImageView? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            val tr = ChangeBounds()
            window.sharedElementExitTransition = tr
        }
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        sharedPreferenceslikes = this@MainActivity.getSharedPreferences(PREFSNAME, 0)

        sharedPreferencesapp = PreferenceManager.getDefaultSharedPreferences(this@MainActivity)
        val notificationstate = sharedPreferencesapp!!.getBoolean(resources.getString(R.string.notificaçãoes), true)
        if (notificationstate) {
            FirebaseMessaging.getInstance().subscribeToTopic("todos")
        } else {
            FirebaseMessaging.getInstance().unsubscribeFromTopic("todos")
        }



        searchViewp.hint = "LuxVilla: Todas"
        searchViewp.setTextSize(18f)
        searchViewp.setVoice(true)
        searchViewp.setTextStyle(1)
        searchViewp.setCursorDrawable(R.drawable.cursor)
        searchViewp.shouldClearOnClose = true

        val nightmode = sharedPreferencesapp!!.getBoolean(resources.getString(R.string.night_mode), false)
        if (nightmode) {
            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
            searchViewp.setTheme(SearchView.THEME_DARK)
            searchViewp.setBackgroundColor(ContextCompat.getColor(this@MainActivity, R.color.card_background))
            searchViewp.setIconColor(ContextCompat.getColor(this@MainActivity, R.color.colorsearchicons))
        } else {
            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
            searchViewp.setTheme(SearchView.THEME_LIGHT)
            searchViewp.setIconColor(ContextCompat.getColor(this@MainActivity, R.color.colorsearchicons))
        }


        tabs.setSelectedTabIndicatorColor(ContextCompat.getColor(this@MainActivity, R.color.colorAccent))
        adapter = BDAdapter(this)


        tabs.setupWithViewPager(vpgr)


        val numerocolunas = adapter!!.numerodecolunas()
        if (numerocolunas == 0 && !isNetworkAvailable(this)) {
            vpgr.visibility = View.GONE
            tabs.visibility = View.GONE
        } else {
            val adaptador = adaptadorpaginas(supportFragmentManager)
            vpgr.adapter = adaptador

            vpgr.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
                override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {

                }

                override fun onPageSelected(position: Int) {

                    when (position) {
                        0 -> {
                            searchViewp.hint = "LuxVilla: Todas"
                            navigationview.menu.getItem(0).isChecked = true
                        }
                        1 -> {
                            searchViewp.hint = "LuxVilla: Aveiro"
                            navigationview.menu.getItem(1).isChecked = true
                        }
                        2 -> {
                            searchViewp.hint = "LuxVilla: Braga"
                            navigationview.menu.getItem(2).isChecked = true
                        }
                        3 -> {
                            searchViewp.hint = "LuxVilla: Porto"
                            navigationview.menu.getItem(3).isChecked = true
                        }
                    }
                    if (!isNetworkAvailable(this@MainActivity)) {
                        Snackbar.make(vpgr, "Sem ligação há internet", Snackbar.LENGTH_LONG).setAction("ligar") {
                            val defenicoes = Intent(Settings.ACTION_WIFI_SETTINGS)
                            startActivity(defenicoes)
                        }.setActionTextColor(ContextCompat.getColor(this@MainActivity, R.color.colorAccent)).show()
                    }
                }

                override fun onPageScrollStateChanged(state: Int) {

                }
            })
        }

        sugestions = ArrayList()
        msearchHistoryTable = SearchHistoryTable(this@MainActivity)
        msearchHistoryTable!!.setHistorySize(3)
        searchViewp.setOnQueryTextListener(object : com.lapism.searchview.SearchView.OnQueryTextListener {
            override fun onQueryTextChange(newText: String): Boolean {
                return false
            }

            override fun onQueryTextSubmit(query: String): Boolean {
                sugestions!!.add(SearchItem(query))
                msearchHistoryTable!!.addItem(SearchItem(query))


                val search = Intent(this@MainActivity, searchableactivity::class.java)
                search.putExtra("query", query)
                startActivity(search)
                searchViewp.shouldClearOnClose = true
                searchViewp.close(true)
                return true
            }
        })
        val searchAdapter = SearchAdapter(this@MainActivity, sugestions)
        searchAdapter.addOnItemClickListener { view, position ->
            val textView = view.findViewById<View>(R.id.textView_item_text) as TextView
            val query = textView.text.toString()

            val search = Intent(this@MainActivity, searchableactivity::class.java)
            search.putExtra("query", query)
            startActivity(search)
            searchViewp.close(true)
        }
        searchViewp.adapter = searchAdapter
        searchViewp.setOnMenuClickListener { drawerll.openDrawer(navigationview) }

        searchViewp.setOnOpenCloseListener(object : SearchView.OnOpenCloseListener {
            override fun onClose(): Boolean {
                val id = vpgr.currentItem
                when (id) {
                    0 -> searchViewp.hint = "LuxVilla: Todas"
                    1 -> searchViewp.hint = "LuxVilla: Aveiro"
                    2 -> searchViewp.hint = "LuxVilla: Braga"
                    3 -> searchViewp.hint = "LuxVilla: Porto"
                }
                searchViewp.setTextStyle(1)
                return true
            }

            override fun onOpen(): Boolean {
                searchViewp.hint = resources.getString(R.string.app_hint)
                searchViewp.setTextStyle(0)
                return true
            }
        })

        val headerLayout = navigationview.getHeaderView(0)
        txtuser=headerLayout.findViewById(R.id.textviewusername)
        txtmail=headerLayout.findViewById(R.id.textviewusermail)
        profileimg=headerLayout.findViewById(R.id.profileimage)

        navigationview.setNavigationItemSelectedListener { item ->
            when (item.itemId) {
                R.id.navigation_item_1 -> vpgr.currentItem = 0
                R.id.navigation_item_2 -> vpgr.currentItem = 1
                R.id.navigation_item_3 -> vpgr.currentItem = 2
                R.id.navigation_item_4 -> vpgr.currentItem = 3
                R.id.navigation_subheader_1 -> {
                    val it = Intent(this@MainActivity, settings::class.java)
                    startActivity(it)
                }
                R.id.navigation_subheader_profile -> startActivity(Intent(this@MainActivity, Userprofile::class.java))
                R.id.navigation_subheader_signout -> {
                    sharedPreferenceslikes!!.edit().clear().apply()
                    FirebaseAuth.getInstance().signOut()
                    startActivity(Intent(this@MainActivity, Loginactivity::class.java))
                    finish()
                }
            }
            drawerll.closeDrawers()
            true
        }


        val shortcutManager: ShortcutManager?

        val shortcut: ShortcutInfo
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N_MR1) {
            shortcutManager = getSystemService(ShortcutManager::class.java)

            shortcut = ShortcutInfo.Builder(this, "id1")
                    .setShortLabel(resources.getString(R.string.busca))
                    .setLongLabel(resources.getString(R.string.busca))
                    .setIcon(Icon.createWithResource(this@MainActivity, R.drawable.ic_shortcut_search))
                    .setIntents(
                            arrayOf(Intent(Intent.ACTION_MAIN, Uri.EMPTY, this, MainActivity::class.java).setFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK), Intent(Intent.ACTION_MAIN, Uri.EMPTY, this, searchableactivity::class.java)))
                    .build()
            val shortcutInfoLink = ShortcutInfo.Builder(this, "shortcut_web")
                    .setShortLabel("website")
                    .setLongLabel("LuxVilla website")
                    .setIcon(Icon.createWithResource(this, R.drawable.logo))
                    .setIntent(Intent(Intent.ACTION_VIEW, Uri.parse("http://brunoferreira.esy.es/")))
                    .build()


            shortcutManager!!.dynamicShortcuts = Arrays.asList(shortcut, shortcutInfoLink)
        }

    }


    public override fun onDestroy() {
        super.onDestroy()
    }


    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)

        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        return super.onOptionsItemSelected(item)
    }

    private inner class adaptadorpaginas internal constructor(fm: FragmentManager) : FragmentPagerAdapter(fm) {

        override fun getItem(position: Int): Fragment? {

            when (position) {
                SEPARADOR_TODAS -> return separadortodas()
                SEPARADOR_AVEIRO -> return separadoraveiro()
                SEPARADOR_BRAGA -> return separadorbraga()
                SEPARADOR_PORTO -> return separadorporto()
            }
            return null
        }

        override fun getPageTitle(position: Int): CharSequence {
            val tabsname = arrayOf("TODAS", "AVEIRO", "BRAGA", "PORTO")

            return tabsname[position]
        }

        override fun getCount(): Int {
            return 4
        }
    }

    public override fun onResume() {
        super.onResume()
        firebaseutils.getuserdata(this@MainActivity, txtuser!!, txtmail!!, profileimg!!)
        if (drawerll.isDrawerOpen(navigationview)) {
            drawerll.closeDrawer(navigationview)
        }
    }

    override fun onRestart() {
        super.onRestart()
        if (android.os.Build.VERSION.SDK_INT <= Build.VERSION_CODES.KITKAT) {
            super.recreate()
        }
    }

    override fun onBackPressed() {

        if (drawerll.isDrawerOpen(navigationview)) {
            drawerll.closeDrawer(navigationview)
        } else {
            super.onBackPressed()
        }

    }

    fun isNetworkAvailable(context: Context): Boolean {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        return connectivityManager.activeNetworkInfo != null && connectivityManager.activeNetworkInfo.isConnected
    }
}
